Use new validation for creating/updating triggers

Dimedrolity opened the pull request:
After https://github.com/moira-alert/moira/pull/817

Dimedrolity committed:
test: cover trigger `save` method

Dimedrolity committed:
Call new version of API with validation.

Now triggers with error functions in targets (a.k.a. red functions) would not be saved.

Commit:
Fix CI - bump ubuntu to 22.04

Commit:
Fix CI Bump version of python in test workflow

Commit:
Fix CI

Dimedrolity commented on .github/workflows/test.yml:
```diff
@@ -7,7 +7,7 @@ jobs:
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.8, 3.9, 3.10, 3.11]
```
Do we not need support 3.6 and 3.7?

kotbauk replied:
Github doesn't support 3.6 and 3.7 in ubuntu-22.04 vm

Dimedrolity commented on .github/workflows/test.yml:
```diff
@@ -7,7 +7,7 @@ jobs:
     strategy:
       matrix:
-        python-version: [3.8, 3.9, 3.10, 3.11]
+        python-version: [3.8, 3.9, 3.10.10, 3.11]
```
Seems strange but it works

kotbauk replied:
Github convert 3.10 to 3.1, so 3.10 doesn't work (probably 3.10.10 converted to 3.10.1 bl bl)

Dimedrolity committed:
Upd version to 4.0.0.

I think it is major release because there is breaking changes. If user now will be used client with trigger with error function, trigger will not be saved, so it is not compatible.

Tetrergeru commented on moira_client/models/trigger.py:
```diff
@@ -149,13 +150,13 @@ def _send_request(self, trigger_id=None):
 
         if trigger_id and api_response:
-            res = self._client.put('trigger/{id}'.format(id=trigger_id), json=data)
+            res = self._client.put('trigger/{id}?{validate}'.format(id=trigger_id, validate=self.QUERY_PARAM_VALIDATE_FLAG), json=data)
```
```suggestion
            res = self._client.put(f'trigger/{trigger_id}?{self.QUERY_PARAM_VALIDATE_FLAG}', json=data)
```

Dimedrolity replied:
Thx, done https://github.com/moira-alert/python-moira-client/pull/26/commits/f5b2e119e3bf344fdcef777da764954b2117385f

Dimedrolity committed:
refactor: fstring

Dimedrolity committed:
Revert 3.6 and 3.7 to pass Github checks to merge PR
